### PR TITLE
Fixes vscode #65464

### LINKED ISSUE #65464: Emmet CSS # character
- VSCode Version: 1.30.1
- OS Version: Windows 10 Pro x64

Emmet doesn't works correct after trying to type character **#**.

Steps to Reproduce:

1. Type `bd1#s`.
2. Press **Tab**.

Emmet must type `border: 1px #000 solid;`, but it doesn't.

emmet-css

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes https://github.com/microsoft/vscode/issues/65464

### CODE REVIEW

File: src/emmetHelper.ts [hunk 1]
```diff
@@ -495,7 +495,11 @@ export function isAbbreviationValid(syntax: string, abbreviation: string): boole
 			return false;
 		}
 		if (abbreviation.includes('#')) {
-			return hexColorRegex.test(abbreviation) || propertyHexColorRegex.test(abbreviation);
```
Comment (rzhao271) [inline src/emmetHelper.ts:498]: I wonder whether `propertyHexColorRegex` should still show up. For example, `bgc:#333` is valid (and expands to `background-color: #333;`).

Comment (jeanp413) [reply src/emmetHelper.ts:498]: There are tests for that case and they still pass
https://github.com/microsoft/vscode-emmet-helper/blob/4fa257726c3c1ceff1b403c21960b2ad4678c2f4/src/test/emmetHelper.test.ts#L713-L721

File: src/emmetHelper.ts [hunk 2]
```diff
@@ -495,7 +495,11 @@ export function isAbbreviationValid(syntax: string, abbreviation: string): boole
...
+			if (abbreviation.startsWith('#')) {
+				return hexColorRegex.test(abbreviation);
+			} else if (commonlyUsedTags.includes(abbreviation.substring(0, abbreviation.indexOf('#')))) {
+				return false;
```
Comment (rzhao271) [inline src/emmetHelper.ts:501]: Not sure what this case is for, considering we're in a `if (isStyleSheet(syntax))` block

Comment (jeanp413) [reply src/emmetHelper.ts:501]: @rzhao271 The whole `if (abbreviation.includes('#'))` block was added as a fix for this https://github.com/microsoft/vscode/issues/56082 where emmet completions would show for id selectors and that's not desired. `propertyHexColorRegex` regex is too restrictive causing expressions like `bd1#s` not being expanded so the fix is to not show completions only for id selectors of the form `<element>#<id>`

**Review by rzhao271 — APPROVED** [review]

**Comment by rzhao271** [comment]
Thanks again! Merging

**Pull request merged**